File: IT/1/Stringhe/Esercizi/Python/esercizio24.py
# file esercizio24.py
# -*- coding: utf-8 -*-

from __future__ import print_function, unicode_literals
from sys import stdin, version_info
import re
if hasattr(__builtins__, 'raw_input'): input = raw_input

print("********************* Cambia la password *********************")
print("La password deve contenere un minimo di otto caratteri, inclusa una maiuscola, un numero e un simbolo speciale.")

pwd = ""
pattern = re.compile(u"^(.{0,7}|[^\\d]*|[^a-zà-ù]*|[^A-ZÀ-Ù]*|[a-zA-ZÀ-ù0-9]*)$")

while not pwd:
	s = input("Digita la nuova password: ")
	if version_info[0] < 3: s = s.decode("utf-8")
	if not pattern.match(s):
		pwd = s
		print("Nuova password salvata.")
	else:
		print("Password non valida!")

File: IT/1/Stringhe/Esercizi/Java/Esercizio24.java
/* file Esercizio24.java */

import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class Esercizio24 {
	public static void main(String [] args) {
		System.out.println("********************* Cambia la password *********************");
		System.out.println("La password deve contenere un minimo di otto caratteri, inclusa una maiuscola, un numero e un simbolo speciale.");
		String pwd = "";

		String r = "^(.{0,7}|[^\\d]*|[^a-zà-ù]*|[^A-ZÀ-Ù]*|[a-zA-ZÀ-ù0-9]*)$";
		Pattern pattern = Pattern.compile(r);

		while (pwd.isEmpty()) {
			String s = System.console().readLine("Digita la nuova password: ");
			if (!pattern.matcher(s).matches()) {
				pwd = s;
				System.out.println("Nuova password salvata.");
			}
			else {
				System.out.println("Password non valida!");
			}
		}
	}
}

